src/cueUtilities.js

```diff
@@ -319,60 +320,83 @@ module.exports = function (params, cy, api) {
-              && cyRenderedPosX <= expandcollapseRenderedEndX + expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY >= expandcollapseRenderedStartY - expandcollapseRenderedRectSize * factor
-              && cyRenderedPosY <= expandcollapseRenderedEndY + expandcollapseRenderedRectSize * factor) {
-              if(opts.undoable && !ur)
-                ur = cy.undoRedo({
-                  defaultActions: false
-                });
-              if(api.isCollapsible(node))
-                if (opts.undoable){
-                  ur.do("collapse", {
-                    nodes: node,
-                    options: opts
-                  });
-                }
-                else
-                  api.collapse(node, opts);
-              else if(api.isExpandable(node))
-                if (opts.undoable)
-                  ur.do("expand", {
-                    nodes: node,
-                    options: opts
-                  });
-                else
-                  api.expand(node, opts);
-
-              // don't draw the cue if the group is expandable, as mouse won't be inside it
-              if (hoveredGroup && !isSelectedGroupsContains(hoveredGroup) && api.isExpandable(node)) {
-                hoveredGroup = null;
-              }
-              refreshCanvasImages();
+          return null;
+        }
+
+        function cueClick(event) {
+          let node;
+          if (hoveredGroup && eventOnCueRegion(hoveredGroup, event)) {
+            node = hoveredGroup;
+          }
+          else if (selectedGroups.length > 0) {
+            node = getGroupOfClickedOnCue(event);
+          }
+
+          if (!node) {
+            return;
+          }
+          let opts = options();
+
+          event.stopPropagation();
+          if(opts.undoable && !ur)
+            ur = cy.undoRedo({
+              defaultActions: false
+            });
+          if(api.isCollapsible(node))
+            if (opts.undoable){
+              ur.do("collapse", {
+                nodes: node,
+                options: opts
+              });
             }
+            else
+              api.collapse(node, opts);
+          else if(api.isExpandable(node))
+            if (opts.undoable)
+              ur.do("expand", {
+                nodes: node,
+                options: opts
+              });
+            else
+              api.expand(node, opts);
+          if (options().appearOnGroupSelect) {
+            // mouse won't be in collapsed group, so shouldn't show cue
+            if (hoveredGroup && api.isExpandable(node)) {
+              hoveredGroup = null;
+            }
+            refreshCanvasImages();
           }
-        });
+        }
+
+        function isClickOnAnyCueRegion(event) {
+          return (hoveredGroup && eventOnCueRegion(hoveredGroup, event)) || (selectedGroups.length > 0 && getGroupOfClickedOnCue(event));
+        }
+
+        function interceptCytoscapeEventsWithinCue(event) {
```

Comment:
I'm not wild about this name since we're not really intecepting a cytoscape event, we're intercepting a normal event to prevent a cytoscape event. Maybe `interceptCueEvent()` would be more in line with what's going on.